modules/api/src/main/java/org/apache/ignite/table/TableView.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.table;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Table view interface.
+ *
+ * @param <R> Table row type.
+ */
+public interface TableView<R> {
+    /**
+     * Return row from table for given key fields.
+     *
+     * @param keyRec Row with key fields set.
+     * @return Row with all fields filled from the table.
+     */
+    public <K> R get(K keyRec);
+
+    /**
+     * Fills given rows with the values from the table.
+     *
+     * @param keyRecs Rows with key fields set.
+     * @return Rows with all fields filled from the table.
+     */
+    public <K> List<R> getAll(List<K> keyRecs);
```

Comment:
I think `List` interface is way too specific for the parameter, and may cause unnecessary conversions and allocations in user code.

It would be great to accept `Iterable<K>` or at least `Collection<K>`.

This applies to all APIs here and below.